Broken link for https://52.demo.plone.org in install/index.md

### Plone Documentation versions affected

- [ ] 5.x
- [X] 6.x

### Description

Tests failed because of a broken link to https://52.demo.plone.org in install/index.md

As documentation is for Plone 6, should this link be removed?

```
(   install/index: line   31) broken    https://52.demo.plone.org/en/ - 502 Server Error: Bad Gateway for url: https://52.demo.plone.org/en/
make: *** [Makefile:192: linkcheckbroken] Error 1
Error: Process completed with exit code 2.
```

--- a/docs/install/index.md
+++ b/docs/install/index.md
@@ -27,7 +27,6 @@ Choose a version.
 
 -   [Plone 6 with Volto frontend](https://demo.plone.org/)
 -   [Plone 6 Classic UI (nightly build)](https://classic.demo.plone.org/login?came_from=/en)
--   [Plone 5.2.x (stable) with Barceloneta frontend](https://52.demo.plone.org/en/)
 :::
 
 :::{grid-item-card} {octicon}`download;1.5em;sd-mr-1` Install

